app/lib/main/run_app.dart

```diff
@@ -106,57 +125,69 @@ Future runFlutterApp() async {
 
   final analytics = Analytics(getBackend());
 
-  listenToAuthStateChanged().listen((currentUser) {
-    if (currentUser?.uid != null) {
-      final sharezoneGateway = SharezoneGateway(
-          authUser: currentUser,
-          memberID: currentUser.uid,
-          references: references);
-
-      final gruppenBeitrittsTransformer = GruppenBeitrittsversuchFilterBloc(
-        einkommendeLinks: dynamicLinkBloc.einkommendeLinks,
-        istGruppeBereitsBeigetreten: (publicKey) async =>
-            await istSchonGruppeMitSharecodeBeigetreten(
-                sharezoneGateway, publicKey),
-      );
+  // Skipping the listeners to the auth state to make the integration tests
+  // work.
+  //
+  // The listeners of "listenToAuthStateChanged().listen()" are not disposed.
+  // Therefore, when signing out, the app still tries to stream the Firestore
+  // documents which results into a "[cloud_firestore/permission-denied] The
+  // caller does not have permission to execute the specified operation."
+  // because the user isn't signed anymore.
+  //
+  // A good solution would be to refactor the listeners. But as a quick
+  // workaround we just skip the listeners for integration tests.
+  if (!isIntegrationTest) {
+    listenToAuthStateChanged().listen((currentUser) {
+      if (currentUser?.uid != null) {
+        final sharezoneGateway = SharezoneGateway(
```

Comment:
Why is `listeners` plural? Do you mean that the `StreamSubscription` is not disposed of correctly? Or a you talking about the Firestore listeners?   

I don't fully understand the explanation.   

Is the following explanation correct?    

The code below that uses `listenToAuthStateChanged` creates `SharezoneGateway` which automatically subscribes to document changes in several collections in Firestore.   
This results in errors if the user logs out since we don't correctly dispose or close the streams inside `SharezoneGateway`.  The errors will cause any integration test to automatically fail.

Currently we just work around that by not executing the code below at all (and thus not creating `SharezoneGateway`). In the future we should ensure that the `SharezoneGateway` does not unnecessarily subscribe to document changes and that we can properly dispose of the subscriptions before logging out.

--------

I still don't fully get why the `  if (!isIntegrationTest) {` fixes it. Don't we also create `SharezoneGateway` for the normal usage of the application? Wouldn't this still lead to the same problems?

Reply:
Discussed with Jonas offline the answers.